packages/metro-config/index.js

```diff
@@ -36,6 +39,53 @@ const INTERNAL_CALLSITES_REGEX = new RegExp(
   ].join('|'),
 );
 
+/**
+ * Resolves the root of an NPM or Yarn workspace, by traversing the file tree upwards from a `candidatePath` in the search for
+ * - a directory with a package.json
+ * - which has a `workspaces` array of strings
+ * - which (possibly via a glob) includes the project root
+ * @param {string} projectRoot Project root to find a workspace root for
+ * @param {string | undefined} candidatePath Current path to search from
+ * @returns Path of a workspace root or `undefined`
+ */
+function getWorkspaceRoot(projectRoot /*: string */, candidatePath /*: string */ = projectRoot) /*: string | void */ {
+  const packageJsonPath = path.resolve(candidatePath, 'package.json');
+  if (fs.existsSync(packageJsonPath)) {
```

Comment:
Since we're reading outside the project there's a reasonable chance of a permissions error - I'd be inclined to treat inaccessible files as effectively non-existent. WDYT?

```suggestion
  if (fs.accessSync(packageJsonPath, fs.constants.R_OK)) {
```

Reply:
It seems `fs.accessSync` doesn't return a boolean like `existsSync` did. I believe I can simply remove this guard and instead make the `catch` below ignore an error with `code === 'ENOENT'`.